src/osp/core/math_int64.h

```diff
@@ -54,25 +54,14 @@ constexpr std::uint64_t absdelta(std::int64_t lhs, std::int64_t rhs) noexcept
 /**
  * @brief (distance between a and b) > threshold
  *
- * This function is quick and dirty, max threshold is limited to 1,431,655,765
  */
-constexpr bool is_distance_near(Vector3l const a, Vector3l const b, std::uint64_t const threshold) noexcept
+constexpr bool is_distance_near(Vector3l const a, Vector3l const b, double const threshold) noexcept
 {
-    std::uint64_t const dx = absdelta(a.x(), b.x());
-    std::uint64_t const dy = absdelta(a.y(), b.y());
-    std::uint64_t const dz = absdelta(a.z(), b.z());
+    double const dx( absdelta(a.x(), b.x()) );
+    double const dy( absdelta(a.y(), b.y()) );
+    double const dz( absdelta(a.z(), b.z()) );
```

Comment:
Personally i don't find switching from assignment initialization to braced initialization easier to read.

Are you doing this to force an error if the type of the result of absdelta is not the same as the type of the variable?

Reply:
I agree. Not particularly a sane move to do this on my end. Curly braces will enforce the type; however, these are those error-prone round braces that have a silly chance of being interpreted as a function declaration. I'm intentionally implicit-casting int64 from absdelta to a double.